src/librustc_mir/const_eval.rs

```diff
@@ -525,7 +525,8 @@ pub fn const_eval_provider<'a, 'tcx>(
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>,
 ) -> ::rustc::mir::interpret::ConstEvalResult<'tcx> {
-    trace!("const eval: {:?}", key);
+    trace!("const eval: {}", key.value.instance);
+    trace!("const eval key: {:?}", key);
```

Comment:
Why trace both `key` and `key.value.instance`?

Reply:
Because the printing via `Display` of the instance goes through the code that was ICEing here before

Reply:
Get it